image_resize.py

```diff
@@ -0,0 +1,40 @@
+import boto3
+import os
+from PIL import Image
+     
+s3_client = boto3.client('s3')
+
+''' 
+Lambda function for thumbnail generation
+
+Gets image from "<bucket>", stores in "<bucket>-resized". Both buckets must exist for smooth execution.
+
+Set thumbnail sizes in THUMBNAIL_SIZES_PX
+
+'''
+THUMBNAIL_SIZES_PX=[100,160,200]
+
+
+     
+def resize_image(original_img_path, resized_img_path, new_size_px):
+    with Image.open(original_img_path) as image:
+        ratio =  max(image.size) / float(new_size_px)
+        image.thumbnail(tuple( int(x / ratio) for x in image.size))
+        image.save(resized_img_path)
+     
+def lambda_handler(event, context):
+    for record in event['Records']:
+        bucket = record['s3']['bucket']['name']
+        key = record['s3']['object']['key'] 
+        original_img = '/tmp/original-{}'.format(key)
+        temp_img = '/tmp/temp-{}'.format(key)
+
+        print("bucket: " + str(bucket))
+        print("key: "+str(key))
+
+        s3_client.download_file(bucket, key, original_img)
+
+        for size_px in THUMBNAIL_SIZES_PX:
+            resize_image(original_img, temp_img, size_px)
+            new_name_key = key.replace(".jpg", "_"+str(size_px)+".jpg")
+            s3_client.upload_file(temp_img, '{}-resized'.format(bucket), new_name_key )
```

Comment:
This approach is strange to me. It seems that you would like to add this file (which is a part of different repo) in current repo. I think that this should be downloaded only to your circleci environment and there used _sed_ to substitute parameters where needed.

Reply:
Oh. I see. I guess I just renamed this file. Otherwise I didn't edit existing code.

Reply:
Interesting thought. In the sense of RADON and my opinion, the orchestrator will get CSAR from the IDE or from the CI/CD tool, which would take the same version that IDE creates, but alter it with some configs. Like you did.

Currently I see the *yaml file as one big config. Sure it should be improved. But as this is a kind of config, I would understand that CI should not modify the file in repo, but just alter it before passing it to the orchestrator. Is this possible? Should we use some other think as "submodule"?